src/vs/workbench/services/progress/browser/progressService.ts

```diff
@@ -68,8 +68,14 @@ export class ProgressService extends Disposable implements IProgressService {
 			return handleStringLocation(location);
 		}
 
+		const isDoNotDisturbEnabled = this.notificationService.getDoNotDisturbMode();
+
 		switch (location) {
 			case ProgressLocation.Notification:
+				// Show progress in status bar if do not disturb mode is enabled
+				if (isDoNotDisturbEnabled) {
+					return this.withWindowProgress({ ...options, location: ProgressLocation.Window }, task);
```

Comment:
I think a much easier solution would be to conditionally set `silent: true` if DND mode is on. Silent notifications that are progress will show in the status bar until clicked 🚀